src/games/rcll/production.clp

```diff
@@ -485,11 +489,26 @@
 (defrule production-mps-broken
 	"The MPS is BROKEN. Inform the referee and reset the machine to stop the conveyor belt."
   (gamestate (state RUNNING) (phase PRODUCTION) (game-time ?gt))
-  ?m <- (machine (name ?n) (state BROKEN) (team ?team) (broken-reason ?reason) (broken-since 0.0))
+  ?m <- (machine (name ?n) (state BROKEN) (team ?team) (broken-reason ?reason) (broken-since 0.0)
+                 (bases-added ?ba) (bases-used ?bu) (cs-retrieved ?cap-loaded))
   =>
   (printout t "Machine " ?n " broken: " ?reason crlf)
   (assert (attention-message (team ?team) (text ?reason)))
   (modify ?m (broken-since ?gt))
+  ; Revoke points awarded for unused bases at RS slide
+  (bind ?unused-bases (- ?ba ?bu))
+  (if (> ?unused-bases 0) then
+   (bind ?deduct (* -1 ?unused-bases))
+   (assert (points (game-time ?gt) (team ?team) (phase PRODUCTION)
+                   (points (* ?deduct ?*PRODUCTION-POINTS-ADDITIONAL-BASE*))
+                   (reason (str-cat "Deducting unused additional bases at " ?n))))
+  )
+  ; Revoke points awarded for an unused cap at CS
+  (if ?cap-loaded then
+    (assert (points (game-time ?gt) (team ?team) (phase PRODUCTION)
+                    (points (* -1  ?*PRODUCTION-POINTS-RETRIEVE-CAP*))
+                    (reason (str-cat "Deducting retrieved cap at " ?n))))
```

Comment:
This should be responsible for deducting the point given for unused retrieved cap point. Did you test it and it didn't work?

Reply:
Right (I tested the code without looking into it again). But it doesn't work because `cs-retrieved` is already reset before:
https://github.com/robocup-logistics/rcll-refbox/blob/ee8ded406b0baa61349e94d944aa15f9bd333341/src/games/rcll/production.clp#L182-L185

It probably makes sense to not modify `cs-retrieved` at that point, it is reset later on anyway.

Reply:
O that's a great catch! this never showed up for me when I tested since I either sent a reset msg or sent the wrong instruction a bit too soon (before workpiece is retrieved). Both of which never triggered this path. Thanks for the catch!
I applied your suggestions. Of course, It makes no sense at all to reset any machine slots while breaking a machine. Apparently, that was the only case this ever happens.
Please, have another look!